src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -884,6 +888,67 @@ class BookingService(
       )
     }
 
+    if (booking.service == ServiceName.approvedPremises.value && booking.application != null && user != null) {
+      val dateTime = OffsetDateTime.now()
+
+      val domainEventId = UUID.randomUUID()
+
+      val offenderDetails = when (val offenderDetailsResult = offenderService.getOffenderByCrn(booking.crn, user.deliusUsername, user.hasQualification(UserQualification.LAO))) {
+        is AuthorisableActionResult.Success -> offenderDetailsResult.entity
+        is AuthorisableActionResult.Unauthorised -> throw RuntimeException("Unable to get Offender Details when creating Booking Cancelled Domain Event: Unauthorised")
+        is AuthorisableActionResult.NotFound -> throw RuntimeException("Unable to get Offender Details when creating Booking Cancelled Domain Event: Not Found")
```

Comment:
I wonder if we should just return null here if we can't access Delius (and maybe log upstream?), as it's only fetching the NOMS number, which we can get at a later date. Doesn't make sense to block a user action for something which is not necessarily their fault.